packages/react-components/src/components/UserGuide/UserGuide.tsx

```diff
@@ -0,0 +1,182 @@
+import { FC, PropsWithChildren, useEffect, useRef, useState } from 'react';
+
+import { FloatingPortal } from '@floating-ui/react';
+import { autoUpdate, flip, shift, useFloating } from '@floating-ui/react-dom';
+import cx from 'clsx';
+
+import cursorImage from '../../stories/assets/cursor.svg';
+
+import { IUserGuide } from './types';
+import VirtualReference from './virtualElementReference';
+
+import styles from './UserGuide.module.scss';
+
+const baseClass = 'user-guide';
+
+const virtualReference = (element: Element, padding: number = 0) =>
+  new VirtualReference(element, padding);
+
+export const UserGuide: FC<PropsWithChildren<IUserGuide>> = ({
+  className,
+  children,
+  cursorPosition = 'bottom',
+  cursorTiming = 'fast2',
+  parentElementName,
+  isVisible = false,
+  elementStyles,
+  zIndex,
+}) => {
+  const [parentElement, setParentElement] = useState<Element | null>(null);
+  const [rect, setRect] = useState<DOMRect | null>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const { refs, x, y, strategy, placement, update } = useFloating({
+    middleware: [shift(), flip()],
+    placement: cursorPosition,
+    open: true,
+    whileElementsMounted: autoUpdate,
+  });
+
+  const handleViewportChange = () => {
+    if (parentElement) {
+      setRect(
+        virtualReference(parentElement).getBoundingClientRect() as DOMRect
+      );
+    }
+  };
+
+  useEffect(() => {
+    if (parentElement !== null) {
+      window.addEventListener('resize', handleViewportChange);
+      window.addEventListener('scroll', handleViewportChange);
+
+      return () => {
+        window.removeEventListener('resize', handleViewportChange);
+        window.removeEventListener('scroll', handleViewportChange);
+      };
+    }
+  }, [parentElement]);
+
+  useEffect(() => {
+    const handleResize = () => update();
+    window.addEventListener('resize', handleResize);
+
+    return () => window.removeEventListener('resize', handleResize);
+  }, [update]);
+
+  useEffect(() => {
+    if (parentElementName) {
```

Comment:
I wonder if we should allow to have optional props for `parentElementName`? Is tour without highlighting elements even an option and can happen? 🤔

Reply:
Explained above